lib/services/service.ex

```diff
@@ -187,15 +187,11 @@ defmodule Services.Service do
 
     removed_dates = parse_listed_dates(removed_dates)
 
-    (dates ++ parse_listed_dates(added_dates))
-    |> Enum.reject(fn date ->
-      # if valid_days is an empty array, the service's dates are likely those in
-      # added_dates, so we can ignore evaluating the day of the week here
-      Enum.member?(removed_dates, date) ||
-        if valid_days != [], do: Timex.weekday(date) not in valid_days
-    end)
-    |> Enum.map(&Timex.to_date/1)
-    |> Enum.uniq()
+    (dates
+     |> Enum.reject(fn date -> Enum.member?(removed_dates, date) end)
+     |> Enum.reject(fn date -> Timex.weekday(date) not in valid_days end)
```

Comment:
Credo is going to complain about the double reject. Probably easiest to make it a private function that does both checks.

Reply:
Ah yep - I saw that - I used `Stream` and Credo stopped complaining 🙊